src/main/java/school/lemon/changerequest/java/generics/PairUtil.java

```diff
@@ -0,0 +1,53 @@
+package school.lemon.changerequest.java.generics;
+
+/**
+ * Created by lera on 20.12.16.
+ */
+public final class PairUtil {
+
+    public static <K, V> boolean equals(Pair<K, V> p1, Pair<K, V> p2) {
+        return p1.getKey().equals(p2.getKey()) && p1.getValue().equals(p2.getValue());
+    }
+
+    public static <K extends Comparable<K>, V> int compareTo(Pair<K, V> p1, Pair<K, V> p2) {
+        return p1.getKey().compareTo(p2.getKey());
+    }
+
+    public static <K, V> V[] getValues(Pair<K, V>[] pairs) {
+        V[] array = (V[]) new Object[pairs.length];
+        int i = 0;
+        for (Pair p : pairs) {
+            array[i] = (V) p.getValue();
+            i++;
+        }
+        return array;
+    }
+
+    public static <K, V> K[] getKeys(Pair<K, V>[] pairs) {
+        K[] array = (K[]) new Object[pairs.length];
+        int i = 0;
+        for (Pair p : pairs) {
+            array[i] = (K) p.getKey();
+            i++;
+        }
+        return array;
+    }
+
+    public static <K extends Comparable, V> int countGreaterThan(Pair<K, V>[] pairs, K element) {
```

Comment:
Please add generic type to `Comparable`. Thus you can avoid type cast here `element.compareTo((K) p.getKey()) == -1`